### PR TITLE
Fix tests on OS X (#43) and set up Travis CI

### PR DESCRIPTION
**Pull request by mehaase**
Fix breaking MacOS tests and set up Travis CI

**This PR is going to be open for a bit while I make sure the build integration is working. I'll ping people when I'm ready for a review.**

### CODE REVIEW

**Comment by mehaase** [comment]
Phew, okay I have a working build for Python 3.6 and Python 3.7 on Ubuntu.

Adding MacOS builds [looks difficult](https://docs.travis-ci.com/user/multi-os/)… to say the least. It looks like the Mac build VM wouldn't have Python3 installed and would need a custom install script to add it.

Windows pre-release support was [announced yesterday](https://blog.travis-ci.com/2018-10-11-windows-early-release). I know the genesis of this ticket was the failing tests on MacOS, but I'm wondering how much effort to put into running MacOS tests right now versus focusing on documentation and closing up open issues. Eventually Travis will make this multi-platform testing easier.

Thoughts?

File: setup.py [hunk 1]
```diff
@@ -28,21 +28,29 @@
         'Intended Audience :: Developers',
         'Topic :: Software Development :: Libraries',
         'License :: OSI Approved :: MIT License',
-        'Programming Language :: Python :: 3.5',
```
Comment (belm0) [inline setup.py]: what was the issue with 3.5 support?

**Comment by belm0** [comment]
I have experience with Python installs on OS X.  It's not difficult, but I don't think this PR should be blocked on it.  It's enough that I can run tests locally when developing changes.

**Comment by belm0** [comment]
on OS X, test run now hangs right from the start:
```
$ PYTHONPATH=. python3 -m pytest tests/
=============================== test session starts ================================
platform darwin -- Python 3.6.5, pytest-3.8.1, py-1.6.0, pluggy-0.7.1
rootdir: /Users/john/dev/trio/trio-websocket, inifile: pytest.ini
plugins: trio-0.5.0, pylint-0.12.3, faulthandler-1.5.0, cov-2.6.0
collected 21 items

tests/test_connection.py ...
```

**Comment by njsmith** [comment]
Yeah, figuring out how to get Travis working from scratch is really painful. That's why I recommend stealing from those who have already done it :-). If you run `pip install cookiecutter; cookiecutter gh:python-trio/cookiecutter-trio` it generates a project directory with fully-set-up Linux/Mac/Windows testing (using Travis+appveyor), that you can steal from...

**Comment by belm0** [comment]
OS X is hanging on `async with connection:` in open_websocket.

https://github.com/HyperionGray/trio-websocket/blob/79452ce195c42d646301cd32854ebeb43c1dc01c/trio_websocket/__init__.py#L47

**Comment by belm0** [comment]
I don't know why `async with connection` would block because WebSocketConnection derives from  AsyncResource, which just returns self on `__aenter__`.

**Comment by njsmith** [comment]
btw, I'm curious to hear how coveralls goes for you... most projects I'm familiar with switched to codecov a few years back, but maybe coveralls has gotten better since then, I wouldn't know because everyone switched :-)

**Comment by belm0** [comment]
Update:  tests hanging seem to be due to using an older wsproto version (0.11.0).  I wonder if it's the same issue @mehaase was hitting with Python 3.5.  I think we should keep 3.5 support if possible, I'll try restoring it after pinning wsproto.

**Comment by belm0** [comment]
Python 3.5 still hanging on Travis after wsproto pin.  Working for me locally on OS X, despite all package dependencies matching.

I suggest keeping the 3.5 support in place, and just keep a bug open to investigate the CI issue.

**Pull request merged**